app/controllers/report_controller.rb

```diff
@@ -7,7 +7,10 @@ class ReportController < ApplicationController
   before_action :redirect_unless_moderator, except: [:create]
 
   def index
-    @reports = Report.where(reviewed: false)
+    @reports ||= []
+    Report.where(reviewed: false).each do |report|
+      @reports << report unless report.item.nil?
+    end
```

Comment:
if the author deletes the post on its own after the report was made.
On second thought it might be better deleting the report on a post/comment-destroy/retract